src/redis_lock/__init__.py

```diff
@@ -238,15 +241,22 @@ def extend(self, expire=None):
         elif error:
             raise RuntimeError("Unsupported error code %s from EXTEND script" % error)
 
-    def _lock_renewer(self, interval):
+    @staticmethod
+    def _lock_renewer(lockref, interval, stop):
         """
         Renew the lock key in redis every `interval` seconds for as long
         as `self._lock_renewal_thread.should_exit` is False.
         """
         log = getLogger("%s.lock_refresher" % __name__)
-        while not self._lock_renewal_thread.wait_for_exit_request(timeout=interval):
+        while not stop.wait(timeout=interval):
             log.debug("Refreshing lock")
-            self.extend(expire=self._expire)
+            lock = lockref()
+            if lock is None:
+                log.debug("The lock no longer exists, "
+                          "stopping lock refreshing")
+                break
+            lock.extend(expire=lock._expire)
+            del lock
```

Comment:
On CPython objects are deleted immediately if refcount is 0 (unlike PyPy, where it might take a while)